custom/staking/src/lib.rs

```diff
@@ -1314,6 +1357,64 @@ decl_module! {
             Self::deposit_event(RawEvent::InvalidatedNominators(caller, expired_nominators));
         }
 
+        /// Enables individual commisions. This can be set only once. Once individual commission
+        /// rates are enabled, there's no going back.  Only Governance committee is allowed to
+        /// change this value.
+        #[weight = SimpleDispatchInfo::FixedOperational(100_000)]
+        fn enable_individual_commissions(origin) {
+            T::RequiredCommissionOrigin::try_origin(origin)
+                .map(|_| ())
+                .or_else(ensure_root)
+                .map_err(|_| Error::<T>::NotAuthorised)?;
+
+            // Ensure individual commissions are not already enabled
+            if let Commission::Global(_) = <ValidatorCommission>::get() {
+                <ValidatorCommission>::put(Commission::Individual);
+                Self::deposit_event(RawEvent::IndividualCommissionInEffect);
+            } else {
+                Err(Error::<T>::AlreadyEnabled)?
+            }
+        }
+
+        /// Changes commission rate which applies to all validators. Only Governance
+        /// committee is allowed to change this value.
+        ///
+        /// # Arguments
+        /// * `new_value` the new commission to be used for reward calculations
+        #[weight = SimpleDispatchInfo::FixedOperational(100_000)]
+        fn set_global_comission(origin, new_value: Perbill) {
+            T::RequiredCommissionOrigin::try_origin(origin)
+                .map(|_| ())
+                .or_else(ensure_root)
+                .map_err(|_| Error::<T>::NotAuthorised)?;
+
+            // Ensure individual commissions are not already enabled
+            if let Commission::Global(old_value) = <ValidatorCommission>::get() {
+                ensure!(old_value != new_value, Error::<T>::NoChange);
+                <ValidatorCommission>::put(Commission::Global(new_value));
+                Self::update_validator_prefs(new_value);
```

Comment:
It might be better if we use a central commission value when the global commission is enabled. This will allow us to not require to update all validators' pref (without their consent).

Reply:
That's how I had it earlier but @adamdossa suggested we update the prefs so that any tooling we use reflects the validator's commission